### PR TITLE
add a Travis build

### PR DESCRIPTION
**Pull request by SethTisue**
and bump versions of stuff while we're at it

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+sudo: false
+language: scala
+jdk: oraclejdk8
+script:
+- git clone -b 2.12.x https://github.com/scala/scala
```
Comment (dwijnand) [inline .travis.yml:5]: for extra speed you could add a `--single-branch` and a `--depth 1` there too :-)

Comment (SethTisue) [reply .travis.yml:5]: oh, good idea (note that `--depth 1` implies `--single-branch`)

Comment (SethTisue) [reply .travis.yml:5]: oh, not a good idea actually ;-)

it needs the history, the commands it issues are `git log` and `git show`

**Comment by SethTisue** [comment]
the Travis build doesn't really test much, just that it runs, but it's better than nothing

**Comment by SethTisue** [comment]
I re-ran Travis a few times to make sure that the caching stuff appears to be working (not re-downloading everything and not re-caching each time either), and it does.

review by @lrytz

File: .travis.yml (hunk 1 above)
Comment (SethTisue) [reply .travis.yml:5]: I guess we could clone it as a "bare" repo, but whatevs.

**Comment by SethTisue** [comment]
this was a mentally soothing election-night project

**Pull request merged**